File: CStoreManagement/app/src/main/java/com/c_store/zhiyazhang/cstoremanagement/presenter/contract/ContractTypeAdapter.java
package com.c_store.zhiyazhang.cstoremanagement.presenter.contract;

import android.graphics.Color;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.c_store.zhiyazhang.cstoremanagement.R;
import com.c_store.zhiyazhang.cstoremanagement.bean.ContractTypeBean;
import com.c_store.zhiyazhang.cstoremanagement.utils.onclick.RecyclerOnItemClickListener;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by zhiya.zhang
 * on 2017/5/10 8:56.
 */

public class ContractTypeAdapter extends RecyclerView.Adapter<ContractTypeAdapter.ViewHolder> {


    public List<ContractTypeBean> ctbs;
    private RecyclerOnItemClickListener onClick;

    void setOnItemClickLitener(RecyclerOnItemClickListener onClick) {
        this.onClick = onClick;
    }

    ContractTypeAdapter(List<ContractTypeBean> ctbs) {
        this.ctbs = ctbs;
    }


    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.contract_type_item, parent, false));
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, int position) {
        holder.type.setText(ctbs.get(position).getTypeName());
        holder.inventory.setText(Integer.toString(ctbs.get(position).getInventory()));
        holder.tonightCount.setText(Integer.toString(ctbs.get(position).getTonightCount()));
        holder.todayCount.setText(Integer.toString(ctbs.get(position).getTodayCount()));
        if (ctbs.get(position).isChangeColor()){
            holder.itemView.setBackgroundColor(Color.YELLOW);
        }else {
            holder.itemView.setBackgroundColor(Color.WHITE);
        }
        if (onClick != null) {
            holder.contract_item.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    onClick.onItemClick(holder.itemView, holder.getLayoutPosition());
                }
            });
            holder.contract_item.setOnLongClickListener(new View.OnLongClickListener() {
                @Override
                public boolean onLongClick(View v) {
                    onClick.onItemLongClick(holder.itemView, holder.getLayoutPosition());
                    return true;
                }
            });
        }
    }

    @Override
    public int getItemCount() {
        return ctbs.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.contract_item)
        LinearLayout contract_item;
        @BindView(R.id.type)
        TextView type;
        @BindView(R.id.inventory)
        TextView inventory;
        @BindView(R.id.tonightCount)
        TextView tonightCount;
        @BindView(R.id.todayCount)
        TextView todayCount;

        ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: KotlinCStoreManagement/app/src/main/java/com/cstore/zhiyazhang/cstoremanagement/presenter/contract/ContractTypeAdapter.kt
package com.cstore.zhiyazhang.cstoremanagement.presenter.contract

import android.annotation.SuppressLint
import android.support.v4.content.ContextCompat
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.cstore.zhiyazhang.cstoremanagement.R
import com.cstore.zhiyazhang.cstoremanagement.bean.ContractTypeBean
import kotlinx.android.synthetic.main.item_contract_type.view.*

/**
 * Created by zhiya.zhang
 * on 2017/6/12 15:10.
 */
class ContractTypeAdapter(val sqliteData: ArrayList<ContractTypeBean>, val isJustLook: Boolean, val ctbs: List<ContractTypeBean>, val listener: (ContractTypeBean) -> Unit) :
        RecyclerView.Adapter<ContractTypeAdapter.ViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder = ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_contract_type, parent, false))

    override fun onBindViewHolder(holder: ViewHolder, position: Int) = holder.bind(ctbs[position], listener, sqliteData.filter { it.typeId == ctbs[position].typeId }.count() != 0, isJustLook)

    override fun getItemCount(): Int = ctbs.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @SuppressLint("SetTextI18n")
        fun bind(ctb: ContractTypeBean, listener: (ContractTypeBean) -> Unit, isRead: Boolean, isJustLook: Boolean) = with(itemView) {
            type.text = ctb.typeName
            if (isJustLook) {
                //一览表
                inventory.text = ctb.modSku.toString() + "/" + ctb.sku.toString()
                tonightCount.text = ctb.todayGh.toString()
                todayCount.text = ctb.todayStore.toString()
                if (ctb.modSku != 0) {
                    type_tag.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.is_edit))
                    type_tag.visibility = View.VISIBLE
                } else {
                    type_tag.visibility = View.GONE
                }
            } else {
                inventory.text = ctb.inventory.toString()
                tonightCount.text = ctb.tonightCount.toString()
                todayCount.text = ctb.todayCount.toString()
                if (ctb.todayStore != 0) {
                    type_tag.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.is_edit))
                    type_tag.visibility = View.VISIBLE
                } else {
                    if (isRead || ctb.isChangeColor) {
                        type_tag.visibility = View.VISIBLE
                        type_tag.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.is_read))
                    } else {
                        type_tag.visibility = View.GONE
                    }
                }
            }
            setOnClickListener {
                listener(ctb)
            }
        }
    }
}